File: Level 1/대충 만든 자판/solution.py
def solution(keymaps, targets):
    answer = []
    key_dict = {}

    for keymap in keymaps:
        for index, key in enumerate(keymap):
            key_dict[key] = (
                min(key_dict[key], index + 1) if key in key_dict else index + 1
            )

    for target in targets:
        count = 0
        for char in target:
            if char not in key_dict:
                count = -1
                break
            count += key_dict[char]
        answer.append(count)

    return answer

File: Level 1/대충 만든 자판/solution.js
function solution(keymaps, targets) {
  const map = {};

  keymaps.forEach((keymap) =>
    keymap
      .split("")
      .forEach(
        (key, index) =>
          (map[key] = map[key] ? Math.min(map[key], index + 1) : index + 1)
      )
  );

  return targets.map(
    (target) => target.split("").reduce((acc, cur) => acc + map[cur], 0) || -1
  );
}
